### PR TITLE
2FA - new tf_totp_secret only if needed

### LINKED ISSUE #160: 2FA - tf_totp_secret will be regenerated after submission
Hi,
Thank for continuing the development!

Each time a user is successfully submitting the second factor, the function "two_factor_token_validation" generates a new "tf_totp_secret" via "complete_two_factor_process". 

This means, it is only possible to login once because the user does not know the newly generated "tf_totp_secret" afterwards.

This bug was introduced with the following commit c8d7adb6172a5ad0e97160344b75d83439bb08a2

### PR DESCRIPTION
**Pull request by malware-watch**
Fixes #160

### CODE REVIEW

**Comment by malware-watch** [comment]
> I looked around a bit - I presume you are using qrcode? which need a constant totp_secret?

Correct. 

It would make sense to regenerate totp_secret if tf_primary_method changes. Maybe the best way would be (is_changing or user.tf_primary_method != primary_method): {generate new totp_secret and update db).

This could be the basis for special cases like if a user does not change tf_primary_method but requests a new qrcode via two_factor_setup.

**Comment by jwag956** [comment]
Ok - this code is really subtle (it really should have been 2 different views IMHO).
So I believe, by tracing the code, that you can be setting the primary_method EITHER because you are logged in and explicitly requesting to change it OR the first time you log in after having setup 2FA.
I agree that it's only the 'changing' option that needs to change totp_secret.

File: flask_security/twofactor.py [hunk 1]
```diff
@@ -141,13 +141,15 @@ def complete_two_factor_process(user, primary_method, is_changing):
      and perform action accordingly
     """
 
-    # We always generate NEW totp secret seems like a good idea and safe to do here.
-    user.tf_totp_secret = generate_totp()
     user.tf_primary_method = primary_method
```
Comment (jwag956) [inline flask_security/twofactor.py]: What would think about
```
if user.tf_primary_method != primary_method:
 user.tf_primary_method = primary_method
```
That way we wouldn't update DB on every successfull/failed 2FA login

Comment (malware-watch) [reply flask_security/twofactor.py]: This works on my testapp.

**Pull request merged**